### PR TITLE
Use Kind to run in-cluster tests

### PR DESCRIPTION
**Pull request by jglick**
Confirmed that [Kind](https://kind.sigs.k8s.io/) can be used to run our tests on ci.jenkins.io.

### CODE REVIEW

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java [hunk 1]
```diff
@@ -474,6 +475,7 @@ public void jnlpWorkingDir() throws Exception {
         r.assertBuildStatusSuccess(r.waitForCompletion(b));
     }
 
+    @Ignore("TODO fails in Kind with 'java.nio.file.AccessDeniedException: /home/jenkins/agent/workspace' from FilePath.mkdirs from FileMonitoringTask$FileMonitoringController.<init>")
```
Comment (Vlatombe) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java]: Caused by https://github.com/kubernetes-sigs/kind/issues/830. I wonder whether it is possible to use the one of the Azure provisioners on `ci.jenkins.io`

Comment (jglick) [reply src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java]: We cannot use any cloud services from here, as they would require authentication.

I will check whether the Rancher provisioner works as in https://github.com/kubernetes-sigs/kind/issues/118#issuecomment-475134086.

File: kind.sh [hunk 2]
```diff
@@ -20,6 +20,12 @@ function cleanup() {
 trap cleanup EXIT
 export KUBECONFIG="$(kind get kubeconfig-path --name $BUILD_TAG)"
 kubectl cluster-info
+
+# https://github.com/kubernetes-sigs/kind/issues/118#issuecomment-475134086
+kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/205a80824bc6ea6095602458c15651f58146eae4/deploy/local-path-storage.yaml
+kubectl patch storageclass local-path -p '{"metadata":{"annotations":{"storageclass.beta.kubernetes.io/is-default-class":"true"}}}'
```
Comment (Vlatombe) [inline kind.sh]:
```suggestion
kubectl annotate storageclass local-path storageclass.beta.kubernetes.io/is-default-class=true
```

**Comment by jglick** [comment]
```
aci-maven-11-s2g10 was marked offline: Connection was broken: java.util.concurrent.TimeoutException: Ping started at 1570550812444 hasn't completed by 1570551052445
```

**Pull request merged**